packages/exportable-manifest/src/index.ts

```diff
@@ -81,10 +82,13 @@ async function makePublishDependency (depName: string, depSpec: string, dir: str
           'because this dependency is not installed. Try running "pnpm install".'
       )
     }
+
+    const semverRangeToken = /^workspace:([^@]+@)?([\^~])$/.exec(depSpec)?.[2] ?? ''
```

Comment:
it might be possible to optimize it further if this will be moved outside of the if statement and then in the if statement you would just check if semverRangeToken was found